src/shogun/kernel/DotKernel.h

```diff
@@ -68,8 +68,19 @@ class CDotKernel : public CKernel
 
 			ASSERT(l->has_property(FP_DOT))
 			ASSERT(r->has_property(FP_DOT))
-			ASSERT(l->get_feature_type() == r->get_feature_type())
-			ASSERT(l->get_feature_class() == r->get_feature_class())
```

Comment:
the old Kernel class only supports computation between instances in the same features class